### PR TITLE
Go dep, local tests, minor refactoring

### PR DESCRIPTION
**Pull request by gabolaev**
When I cloned the repo and made some changes, I realized that to run the tests I have to cmd + c / v all the lines from travis config or run a local travis in the docker (ugh ...) or push my changes to github and to look at the Travis CI (and turn it on at https://travis-ci.org for my local repo)

I decided to add the ability to run tests locally without reference to the continious integration services like Travis CI, Coveralls, etc..

I preferred to start from addition of go `dep` to get rid of errors like this:
``` bash
config_test.go:7:2: cannot find package "github.com/stretchr/testify/assert" in ...
```

For the ability to run tests locally, I added a Makefile, which contains:
``` makefile
init: # dep ensure and installations
up_docker_server: # run test clickhouse-server locally from Yandex official Docker Image
test: # tests, linting, fmt
```
And now .travis.yml calls makefile targets. Also _Development_ section was added at the README file with short instruction about tests running.

Summarizing the above and other changes:

- [x] Go dep
- [x] Makefile
- [x] Quick start local tests
- [x] Minor refactoring
- [x] coverage.out gitignore
- [x] Go 1.11 build job at the Travis CI
- [x] Example code indent optimizations

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+up_docker_server:
+	test -n "$$(docker ps -a --format {{.Names}} | grep -o dbr-clickhouse-server)" && docker rm -f dbr-clickhouse-server; \
+	docker run -p 127.0.0.1:8123:8123 --name dbr-clickhouse-server -d yandex/clickhouse-server;
+
+test:
+	test -z "$$(golint ./... | grep -v vendor | tee /dev/stderr)"
+	go vet ./...
+	test -z "$$(gofmt -d -s $$(find . -name \*.go -print | grep -v vendor) | tee /dev/stderr)"
+	go test -v -covermode=count -coverprofile=coverage.out .
```
Comment (bgaifullin) [inline Makefile]: please add new line at end of file

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,15 @@
+SHELL := /bin/bash
+
+init:
+	dep ensure -v
+	go install ./vendor/...
+
+up_docker_server:
+	test -n "$$(docker ps -a --format {{.Names}} | grep -o dbr-clickhouse-server)" && docker rm -f dbr-clickhouse-server; \
+	docker run -p 127.0.0.1:8123:8123 --name dbr-clickhouse-server -d yandex/clickhouse-server;
```
Comment (bgaifullin) [inline Makefile]: IMHO it would be better to add option --rm=true  to remove container after stop

File: .gitignore [hunk 3]
```diff
@@ -1,2 +1,3 @@
 .idea/
 _vendor/
+vendor/
```
Comment (bgaifullin) [inline .gitignore]: Please add new line at end and add Gopkg.lock file here and remove file Gopkg.lock

File: Gopkg.toml [hunk 4]
```diff
@@ -0,0 +1,34 @@
+# Gopkg.toml example
```
Comment (bgaifullin) [inline Gopkg.toml]: please remove commented example.

File: README.md [hunk 5]
```diff
@@ -54,7 +54,7 @@ import (
 	"database/sql"
 	"log"
 	"time"
-
+
```
Comment (bgaifullin) [inline README.md]: please revert, for empty string indent is not needed

File: Makefile [hunk 6]
```diff
@@ -8,7 +8,7 @@ up_docker_server:
 	test -n "$$(docker ps -a --format {{.Names}} | grep -o dbr-clickhouse-server)" && docker rm -f dbr-clickhouse-server; \
 	docker run -p 127.0.0.1:8123:8123 --name dbr-clickhouse-server -d yandex/clickhouse-server;
 
-test:
+test: up_docker_server
 	test -z "$$(golint ./... | grep -v vendor | tee /dev/stderr)"
 	go vet ./...
 	test -z "$$(gofmt -d -s $$(find . -name \*.go -print | grep -v vendor) | tee /dev/stderr)"
```
Comment (bgaifullin) [inline Makefile]: need to stop container after test in this case

File: Makefile (hunk 2 above)
Comment (gabolaev) [reply Makefile]: wow! i didn't know about that flag

**Comment by bgaifullin** [comment]
I have remove go 1.8 from travis.yml, please rebase to master

**Comment by gabolaev** [comment]
_it's force-push evening today, lol_

**Comment by gabolaev** [comment]
DAMN, i did it again

**Comment by gabolaev** [comment]
> I have remove go 1.8 from travis.yml, please rebase to master

done

**Pull request merged**